### DISCUSSION TITLE
Automatic Resolve "Same Signature Methods" In Abstract Class

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#5775] [General]
Currently Abstract Methods Can Be Declared With Different Generic Arguments, But During Implementations User Might Implement The Same Type For Different Generic Parameters, Therefore Causing Abstract Methods To Be Duplicate.

In such cases, the compiler should be able to resolve this issue by using only one method.

```C#
abstract class<T1,T2> MyAbstractClass
{
    public abstract MethodA(T1 parameter);
    public abstract MethodA(T2 parameter);
    //here they can be declared as two different methods
}
public class MyClass : MyAbstractClass<int,int>
{
    public override MethodA(int parameter);
    //Currently this can't compile, but compiler should use this implementation for the abstract methods
}
```

**Comment by Rand-Random** [comment]
Stumbled accross this issue today as well
https://stackoverflow.com/questions/71307242
